src/providers.jl

```diff
@@ -82,7 +90,6 @@ end
 
 # If we're using MKL, load it in and set library paths appropriately.
 @static if fftw_provider == "mkl"
-    import MKL_jll
```

Comment:
Why not doing the test `is_available` here?  People are not going to be happy to having to unconditionally download MKL, that was the whole point of #133

Reply:
I was not aware of this issue. Mainly, I thought it would be more user-friendly to figure out the supported libraries in advance, before trying to set them. Only moving the check here won't help with my use case and the linked issue above.

Reply:
Is there any other way to check whether a library can be loaded apart from `..._jll.is_available()`? Maybe checking the system architecture is an OKish workaround?

Reply:
> Is there any other way to check whether a library can be loaded apart from ..._jll.is_available()?

No.

> Maybe checking the system architecture is an OKish workaround?

That's going to hard-code answers which may potentially change over time.

Reply:
>  Only moving the check here won't help with [...] the linked issue above.

Why not?  Capture the fact that the artifact isn't available and throw a helpful error message to explain what to do, instead of getting a cryptic `UndefVarError` that some users don't know what to do with.

Reply:
Yeah, switching to the supported default provider was the main point - changing the LocalPreferences.toml file manually on my end creates git conflicts when moving branches and pulling all the time (and if I forget to adjust it, compilation will fail). To some extent that's already done in the current code: If you specify an invalid provider other than `fftw` and `mkl`, FFTW will just display an `@error` and switch to the default provider (which is fixed to `fftw` currently). So basically I'm in the same situation, on my computer `mkl` is also an invalid provider so I would like to switch to the default one. And I guess you could hypothetically think about an analogous scenario where `fftw` is invalid and you would like to switch to the `mkl` as default provider.

Reply:
Displaying a more descriptive error message is an improvement but I think a solution that just defines the (in)valid providers correctly would be much more convenient.

Reply:
> That's going to hard-code answers which may potentially change over time.

Yes, a bit annoying. But since I didn't see any other alternative to define the valid providers correctly, I switched to hardcoding the supported platforms.

I guess the supported platforms could always be updated easily - and probably supported platforms for MKL won't change much at least? If we're more ambitious, probably we could even automatically extract it from the Artifacts.toml files in FFTW_jll and MKL_jll.

Reply:
> I guess the supported platforms could always be updated easily - and probably supported platforms for MKL won't change much at least?

It did change.